astrapy/exceptions.py

```diff
@@ -0,0 +1,83 @@
+import logging
+import httpx
+from typing import Any, Dict, Optional, TypeVar, cast
+
+from astrapy.types import API_RESPONSE
+from astrapy.utils import make_request
+
+T = TypeVar("T", bound="APIRequestHandler")
+
+
+logger = logging.getLogger(__name__)
+
+
+class APIRequestError(ValueError):
+    def __init__(self, response: httpx.Response) -> None:
+        self.response = response
+        super().__init__(response.text)
+
+    def __repr__(self) -> str:
+        return f"{self.response}"
+
+
+class APIRequestHandler:
```

Comment:
nit: the name of the file is "exceptions.py", but it actually contains the code that does the http class, maybe we can have a better name ?

Reply:
I agree. Maybe `api.py`? it's still not exactly the most clear, but since the request handlers and exception classes all are derived and related to making API requests...